apis/provider-ceph/v1alpha1/bucket_types.go

```diff
@@ -104,7 +104,15 @@ type BucketSpec struct {
 	// existing buckets from the existing S3 backends and the Bucket
 	// CR's status is updated accordingly.
 	// This flag overrides 'Providers'.
-	Disabled          bool `json:"disabled,omitempty"`
+	Disabled bool `json:"disabled,omitempty"`
+	// +optional
+	// AutoPause allows the user to disable further reconciliation
+	// of the bucket after successfully created or updated.
+	// The magic under the hood is `crossplane.io/paused` annotation. (https://docs.crossplane.io/latest/concepts/managed-resources/#paused)
+	// If `crossplane.io/paused` is `true`, disables reconciliation of object.
+	// If `crossplane.io/paused`is missing or empty, triggers auto pause function.
```

Comment:
If `crossplane.io/paused` is missing surely we want to disable auto-pause?

Reply:
When it is missing or empty, crossplane would trigger the reconciliation, we run reconciliation loop, but at the end if bucket is healty we set pause annotation. Any other value means auto pause is disabled.